kafe2/fit/xy/cost.py

```diff
@@ -159,336 +128,46 @@ def __init__(self, errors_to_use='covariance', fallback_on_singular=True, axes_t
-                errors_to_use=errors_to_use, fallback_on_singular=fallback_on_singular)
+            self._COV_MAT_INVERSE_NAME = "y_total_cov_mat_inverse"
+            self._ERROR_NAME = "y_total_error"
         elif axes_to_use.lower() == 'xy':
-            _cost_function_description = "chi-square with projected x errors"
-            if errors_to_use is None:
-                _chi2_func = self.chi2_no_errors
-                _cost_function_description += ' (no errors)'
-            elif errors_to_use.lower() == 'covariance':
-                if fallback_on_singular:
-                    _chi2_func = self.chi2_xy_covariance_fallback
-                else:
-                    _chi2_func = self.chi2_xy_covariance
-                _cost_function_description += ' (covariance matrix)'
-            elif errors_to_use.lower() == 'pointwise':
-                if fallback_on_singular:
-                    _chi2_func = self.chi2_xy_pointwise_errors_fallback
-                else:
-                    _chi2_func = self.chi2_xy_pointwise_errors
-                _cost_function_description += ' (pointwise errors)'
-            else:
-                raise CostFunctionException(
-                    "Unknown value '%s' for 'errors_to_use': must be one of ('covariance', 'pointwise', None)")
-            CostFunctionBase.__init__(self, cost_function=_chi2_func)
-            self._formatter.latex_name = "\chi^2"
-            self._formatter.name = "chi2"
-            self._formatter.description = _cost_function_description
-            self._is_chi2 = True
+            self._COV_MAT_INVERSE_NAME = "projected_xy_total_cov_mat_inverse"
+            self._ERROR_NAME = "projected_xy_total_error"
         else:
-            raise CostFunctionException("Unknown value '%s' for 'axes_to_use': must be one of ('xy', 'y')")
+            raise CostFunctionException(
+                "Unknown value '%s' for 'axes_to_use': must be one of ('xy', 'y')")
+        super(XYCostFunction_Chi2, self).__init__(
+            errors_to_use=errors_to_use, fallback_on_singular=fallback_on_singular)
 
     def on_no_errors(self):
         if not self._no_errors_warning_printed:
-            if (self._cost_function_handle is self.chi2_covariance_fallback
-                    or self._cost_function_handle is self.chi2_pointwise_errors_fallback):
-                print('WARNING: No y data errors were specified. Setting y data errors to 1.')
-            elif (self._cost_function_handle is self.chi2_xy_covariance_fallback
-                    or self._cost_function_handle is self.chi2_xy_pointwise_errors_fallback):
-                print('WARNING: No y data errors were specified. Will set y data errors to 1 if total error becomes 0.')
-            else:
-                print('WARNING: No y data errors were specified. The fit results may be wrong.')
+            print('WARNING: No y data errors were specified. Will set y data errors to 1 '
+                  'if total error becomes 0.')
```

Comment:
Needs `import warnings`

```suggestion
            warnings.warn('No y data errors were specified. Will set y data errors to 1 '
                  'if total error becomes 0.')
```